### PR TITLE
feat(operations): add explain endpoint for search

### PR DESCRIPTION
**Pull request by RyanHolstien**
Explain endpoint has a limitation of only accepting a single index, so it isn't a perfect replication of our multi-index query. Still useful, but won't be a perfect tool for seeing total ranking comparison.
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/ESSearchDAO.java [hunk 1]
```diff
@@ -463,4 +465,35 @@ private String createPointInTime(String[] indexArray, String keepAlive) {
...
   }
+
+  public ExplainResponse explain(
+      @Nonnull String query,
+      @Nonnull String documentId,
+      @Nonnull String entityName,
+      @Nullable Filter postFilters,
+      @Nullable SortCriterion sortCriterion,
+      @Nullable SearchFlags searchFlags,
+      int from,
```
Comment (david-leifker) [inline metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/ESSearchDAO.java:476]: Are these parameters used? If so, why not the scroll pagination method instead?

Comment (david-leifker) [reply metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/ESSearchDAO.java:476]: It almost seems like these just don't need to be exposed in the method maybe?

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java [hunk 2]
```diff
@@ -130,4 +146,86 @@ public ResponseEntity<String> getIndexSizes() {
...
+          @Nonnull
+          String documentId,
+      @Parameter(
+              name = "entityName",
+              required = true,
+              description = "Name of the entity the document belongs to.")
+          @RequestParam("entityName")
+          @Nonnull
+          String entityName,
+      @Parameter(name = "from", required = true, description = "Start point for pagination.")
```
Comment (david-leifker) [inline metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java:176]: I guess this is used, but not sure why we're not using the scroll api

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java [hunk 3]
```diff
@@ -130,4 +146,86 @@ public ResponseEntity<String> getIndexSizes() {
...
+          List<String> facets) {
+
+    Authentication authentication = AuthenticationContext.getAuthentication();
+    String actorUrnStr = authentication.getActor().toUrnStr();
+    DisjunctivePrivilegeGroup orGroup =
+        new DisjunctivePrivilegeGroup(
+            ImmutableList.of(
+                new ConjunctivePrivilegeGroup(
+                    ImmutableList.of(
+                        PoliciesConfig.GET_TIMESERIES_INDEX_SIZES_PRIVILEGE.getType()))));
```
Comment (david-leifker) [inline metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java:218]: Timeseries index privilege probably doesn't make sense.

Comment (david-leifker) [reply metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java:218]: ES_EXPLAIN_QUERY_PRIVILEGE

**Review by david-leifker — CHANGES_REQUESTED** [review]
Permissions check and pagination method

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java (hunk 3 above)
Comment (RyanHolstien) [reply metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java:218]: Thanks for the catch, thought I changed this

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java (hunk 2 above)
Comment (RyanHolstien) [reply metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/operations/elastic/OperationsController.java:176]: We can switch it over to scroll if you think that'd be more useful, scroll is still not widely used for search though.

**Review by david-leifker — APPROVED** [review]

**Pull request merged**